akka-http-tests/src/test/scala/akka/http/scaladsl/server/DontLeakActorsOnFailingConnectionSpecs.scala

```diff
@@ -45,7 +45,8 @@ abstract class DontLeakActorsOnFailingConnectionSpecs(poolImplementation: String
       assertAllStagesStopped {
         val reqsCount = 100
         val clientFlow = Http().superPool[Int]()
-        val (host, port) = SocketUtil.temporaryServerHostnameAndPort()
+        val host = "127.0.0.46" // unlikely to be used host / port
```

Comment:
This won't work on non-linux machines right? I'm not sure how this is any better compared to `SocketUtil.temporaryServerHostnameAndPort` though?

Reply:
`SocketUtil.temporaryServerHostnameAndPort` will offer a port that is likely to be used afterwards which will lead to hard to diagnose behavior when other tests are running at the same time.

Maybe there are other addresses that are know to be inaccessible?

Reply:
Thinking about it, the whole idea of this test is to connect to a port that is *unavailable*, so choosing from the localhost block should still work on systems that don't react on those loopback addresses. I guess the detail error message may be different ("not routable" vs. "connection refused"). Can you try if it's problem for you, @raboof?

Reply:
Ah, you probably don't got such a machine either... I guess, let's react when someone complains.

Reply:
Why would ports returned by `SocketUtil.temporaryServerHostnameAndPort` be likely to be used afterwards? 
```
(for (_ ← Range(0, 10000)) yield {
        SocketUtil.temporaryServerHostnameAndPort()
      }).groupBy(identity _).collect { case (pair, s) if s.size > 1 ⇒ pair }
```
This only turned up a duplicate after a number of attempts. I haven't done the math but that seems fairly plausibly random to me (birthday paradox).

That said, it could be nice to reserve some ports for this kind of use and make sure `temporaryServerHostnameAndPort` stays away from those as well.

Reply:
> This only turned up a duplicate after a number of attempts. I haven't done the math but that seems fairly plausibly random to me (birthday paradox).

That's good to hear but not really evidence that it would be the same behavior on any other system. We just don't control the allocation procedure.

> That said, it could be nice to reserve some ports for this kind of use and make sure temporaryServerHostnameAndPort stays away from those as well.

But that would require that the OS also takes part in the reservation otherwise it could still offer addresses to other users. By now I think that using `temporaryServerHostnameAndPort` is an unreliable anti-pattern that we should try to avoid (and replace by using `0` when binding servers etc).

Reply:
Hmm when using an address that isn't known I indeed see a bit weird behavior. It seems the first few connection attempts come back quickly with a "No route to host" error. Afterwards the next connection attempts seem to be stuck a bit longer however eventually running into the connection timeout. I wonder where this behavior comes from.

Reply:
I completely agree about avoiding `temporaryServerHostnameAndPort` and just binding to `0` where possible ;)